### PR TITLE
[Web] Deprecation fixes.

### LINKED ISSUE #307: [Web] `initKeyEvent` function is deprecated.
See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/initKeyEvent - we use this function for an old bugfix/workaround in the code, but it's deprecated.  If we still need it, we should adapt the code accordingly.

### LINKED ISSUE #309: KeyboardEvent.keyCode is deprecated.
See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode.

Our first check for a keyboard event's pressed key code is through this property.  At minimum, it ought be the second check, after the primary recommendation in that article.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #307 and refactors keycode processing to a centralized method (in prep for eventual future work on #309).

Replaces use of deprecated `KeyboardEvent` fields and methods where currently-promoted alternatives are available, defaulting to old behaviors on browsers that do not support the new, 'modern' methods.

Currently marked [WIP] due to ongoing investigation into the necessity of preserving `_FindCaret`.

### CODE REVIEW

File: web/source/keymanweb.js [hunk 1]
```diff
@@ -2817,6 +2817,25 @@ if(!window['keyman']['initialized']) {
...
 
+    /**
+     * Function     _GetEventKeyCode
+     * Scope        Private
+     * @param       {Event}       e         Event object
+     * Description  Finds the key code represented by the event, prioritizing modern methods over deprecated ones.
+     *              Necessary because the 'modern' methods aren't yet fully implemented across all browsers at this time.
+     */
+    keymanweb._GetEventKeyCode = function(e) {
+      if(e.code && keymanweb._codeToLcodeMap[e.code]) {  // Unfortunately, the former DOES NOT map to a nice integer code!
```
Comment (mcdurdin) [inline web/source/keymanweb.js]: I'm not in favour of using `event.code` at present. Support is still inconsistent (IE, Edge, Android Browser, Chrome on Android don't support it - https://caniuse.com/#feat=keyboardevent-code) and I can't see that it actually adds any benefit at this time -- instead bulking out our deployable code with a big map of strings.

File: web/source/keymanweb.js [hunk 2]
```diff
@@ -3003,17 +3023,26 @@ if(!window['keyman']['initialized']) {
...
     {
       if(!Pelem.createTextRange && Pelem.selectionStart)
-      {      
-        var Levent=document.createEvent('KeyboardEvent');
-        if(Levent.initKeyEvent)
-        {
-          Levent.initKeyEvent('keypress',true,true,null,false,false,false,false,0,32);
+      {
+        var Levent = new KeyboardEvent('keypress', {
+          "code":"Space",
```
Comment (mcdurdin) [inline web/source/keymanweb.js]: I don't think this is going to work on IE. We need to stay with `initKeyEvent`. There's no benefit to two code paths here.

File: web/source/keymanweb.js [hunk 3]
```diff
@@ -3003,17 +3023,26 @@ if(!window['keyman']['initialized']) {
...
+        if(Levent) {
           Levent._kmw_block = true; // Tell keymanweb._KeyPress to ignore this simulated event.
           
           // Nasty problems arise from the simulated keystroke if we don't temporarily block the handler.
           Pelem.dispatchEvent(Levent);
-          Levent=document.createEvent('KeyboardEvent');
-          Levent.initKeyEvent('keypress',true,true,null,false,false,false,false,8,0);
+
+          Levent = new KeyboardEvent('keypress', {
+            "key":"Backspace",
```
Comment (mcdurdin) [inline web/source/keymanweb.js]: This won't work on IE, as above.

File: web/source/keymanweb.js [hunk 4]
```diff
@@ -4818,25 +4847,25 @@ if(!window['keyman']['initialized']) {
...
-      if(!e) e = window.event;
-      if (e.keyCode) var _Lcode = e.keyCode;
-      else if (e.which) var _Lcode = e.which;
-      else return 0;
+    keymanweb._ProcessHotKeys = function(e) {
+      if(!e) {
+        e = window.event;
+      }
+
+      _Lcode = keymanweb._GetEventKeyCode(e);
```
Comment (mcdurdin) [inline web/source/keymanweb.js]: `var _Lcode = ....`

Comment (mcdurdin) [reply web/source/keymanweb.js]: Note: this was picked up by the compiler as well.

File: web/source/keymanweb.js (hunk 1 above)
Comment (jahorton) [reply web/source/keymanweb.js]: I've tried searching for documentation on IE, but to no avail.  It's a shame that link couldn't provide a reference link to that 'inconsistent' statement for us.

File: web/source/keymanweb.js (hunk 2 above)
Comment (jahorton) [reply web/source/keymanweb.js]: Good, that's fine.  This was always a Firefox-oriented fix - it's not supposed to run on IE.

File: web/source/keymanweb.js (hunk 3 above)
Comment (jahorton) [reply web/source/keymanweb.js]: Good, as above.

File: web/source/keymanweb.js (hunk 2 above)
Comment (jahorton) [reply web/source/keymanweb.js]: Also, I'm in the process of investigating whether we can outright drop the function containing this code - `_FindCaret` is explicitly a workaround for an old Firefox bug, so it's probably no longer something we need in KMW.

**Pull request merged**